modules/mrz-reader/ios/MrzReaderView.swift

```diff
@@ -0,0 +1,402 @@
+        
+        // Line 2: DocumentNumber<CheckDigit Nationality DOB<CheckDigit Sex Expiry<CheckDigit Optional<CheckDigit
+        let documentNumberRaw = String(line2.prefix(9)) // Includes fillers
+        let documentNumberCheckDigit = line2[line2.index(line2.startIndex, offsetBy: 9)]
+        let documentNumber = documentNumberRaw.replacingOccurrences(of: "<", with: "").trimmingCharacters(in: .whitespaces)
+        
+        let nationality = String(line2[line2.index(line2.startIndex, offsetBy: 10)..<line2.index(line2.startIndex, offsetBy: 13)])
+        let dob = String(line2[line2.index(line2.startIndex, offsetBy: 13)..<line2.index(line2.startIndex, offsetBy: 19)]) // YYMMDD
+        let dobCheckDigit = line2[line2.index(line2.startIndex, offsetBy: 19)]
+        let sex = String(line2[line2.index(line2.startIndex, offsetBy: 20)..<line2.index(line2.startIndex, offsetBy: 21)]) // M, F, or <
+        let expiry = String(line2[line2.index(line2.startIndex, offsetBy: 21)..<line2.index(line2.startIndex, offsetBy: 27)]) // YYMMDD
+        let expiryCheckDigit = line2[line2.index(line2.startIndex, offsetBy: 27)]
+        let optionalData = String(line2[line2.index(line2.startIndex, offsetBy: 28)..<line2.index(line2.startIndex, offsetBy: 42)]).replacingOccurrences(of: "<", with: "").trimmingCharacters(in: .whitespaces)
+        let optionalDataCheckDigit = line2[line2.index(line2.startIndex, offsetBy: 42)]
+        let overallCheckDigit = line2[line2.index(line2.startIndex, offsetBy: 43)] // Composite check digit
+        
+        // Validate check digits
+        guard validateCheckDigit(value: documentNumberRaw, checkDigit: documentNumberCheckDigit),
+              validateCheckDigit(value: dob, checkDigit: dobCheckDigit),
+              validateCheckDigit(value: expiry, checkDigit: expiryCheckDigit),
+              validateCheckDigit(value: optionalData.isEmpty ? "0" : optionalData, checkDigit: optionalDataCheckDigit) else {
+            return nil
+        }
+        
+        // Validate composite check digit (document number + check + dob + check + expiry + check + optional + check)
+        let compositeValue = documentNumberRaw + String(documentNumberCheckDigit) +
+                            dob + String(dobCheckDigit) +
+                            expiry + String(expiryCheckDigit) +
+                            optionalData
+        guard validateCheckDigit(value: compositeValue, checkDigit: overallCheckDigit) else { return nil }
+        
+        // Format dates (YYMMDD to YYYY-MM-DD)
+        let currentYear = Calendar.current.component(.year, from: Date())
+        let century = currentYear / 100
+        let dobYear = Int(dob.prefix(2)) ?? 0
+        let expiryYear = Int(expiry.prefix(2)) ?? 0
+        
+        // If the year is greater than current year + 10, assume previous century
+        let dobCentury = dobYear > (currentYear % 100 + 10) ? century - 1 : century
```

Comment:
It could be simplified to check if dobYear > currentYear, otherwise passport owner would be born in the future

```suggestion
        let dobCentury = dobYear > (currentYear % 100) ? century - 1 : century
```